src/Service.Tests/CosmosTests/QueryFilterTests.cs

```diff
@@ -642,6 +647,135 @@ public async Task TestFilterOnInnerNestedFields()
+                { 
+                    items {
+                        name
+                    }
+                }
+            }";
+            string clientRoleHeader = AuthorizationType.Anonymous.ToString();
+            JsonElement response = await ExecuteGraphQLRequestAsync(
+                queryName: "earths",
+                query: gqlQuery,
+                variables: new() { { "name", "test name" } },
+                authToken: AuthTestHelper.CreateStaticWebAppsEasyAuthToken(specificRole: clientRoleHeader),
+                clientRoleHeader: clientRoleHeader);
+
+            // Validate the result contains the GraphQL authorization error code.
+            string errorMessage = response.ToString();
+            Assert.IsTrue(errorMessage.Contains(DataApiBuilderException.GRAPHQL_FILTER_FIELD_AUTHZ_FAILURE));
+        }
+
+        /// <summary>
+        /// Tests that the field level query filter succeeds requests when filter fileds are authorized
+        /// </summary>
+        [TestMethod]
+        public async Task TestQueryFilterFieldAuth_AuthorizedWildCard()
+        {
+            // Run query
+            string gqlQuery = @"{
+                planets(first: 1, " + QueryBuilder.FILTER_FIELD_NAME + @" : {name : {eq : ""Earth""}})
+                { 
+                    items {
+                        name
+                    }
+                }
+            }";
+            string clientRoleHeader = AuthorizationType.Anonymous.ToString();
+            JsonElement response = await ExecuteGraphQLRequestAsync(
+                queryName: "planets",
+                query: gqlQuery,
+                variables: new() { },
+                authToken: AuthTestHelper.CreateStaticWebAppsEasyAuthToken(specificRole: clientRoleHeader),
+                clientRoleHeader: clientRoleHeader);
+
+            Assert.AreEqual(response.GetProperty("items")[0].GetProperty("name").ToString(), "Earth");
+        }
+
+        /// <summary>
+        /// Tests that the nested field level query filter succeeds requests when nested filter fileds are authorized
+        /// </summary>
+        [TestMethod]
+        public async Task TestQueryFilterNestedFieldAuth_AuthorizedNestedField()
+        {
+            string gqlQuery = @"{
+                planets(first: 1, " + QueryBuilder.FILTER_FIELD_NAME + @" : {earth : {id : {eq : """ + _idList[0] + @"""}}})
+                { 
+                    items {
+                        earth {
+                                id
+                              }
+                    }
+                }
+            }";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(_graphQLQueryName, query: gqlQuery);
+            Assert.AreEqual(actual.GetProperty("items")[0].GetProperty("earth").GetProperty("id").ToString(), _idList[0]);
+        }
+
+        /// <summary>
+        /// Tests that the nested field level query filter fails requests when nested filter fileds are unauthorized
```

Comment:
```suggestion
        /// Tests that the nested field level query filter fails authorization when nested filter fields are 
        /// unauthorized because the field 'name' on object type 'earth' is an excluded field of the read
        /// operation permissions defined for the anonymous role.
```

Reply:
Updated